python/ray/experimental/compiled_dag_ref.py

```diff
@@ -33,18 +33,14 @@ def __init__(
         self._execution_index = execution_index
         # Whether ray.get() was called on this CompiledDAGRef.
         self._ray_get_called = False
-        self._reader_refs = (
-            dag.dag_output_channels._reader_ref
-            if dag.has_single_output
-            else [channel._reader_ref for channel in dag.dag_output_channels]
-        )
+        self._dag_output_channels = dag.dag_output_channels
 
     def __repr__(self):
         return (
             f"CompiledDAGRef(_dag={self._dag}, "
```

Comment:
In case you change the method to `__str__()`, we can remove `_dag={self._dag}` as printing an address is not quite helpful information.